app/scripts/metamask-controller.js

```diff
@@ -416,6 +421,16 @@ module.exports = class MetamaskController extends EventEmitter {
   // Identity Management
   //
 
+  setLocked (cb) {
+    this.keyringController.setLocked()
+      .then((err, result) => {
+        if (err) { cb(err) }
```

Comment:
prolly should be `if (err) return cb(err)`
unless we also want to clear the selected address